{
  "id": "RUSTSEC-2018-0001",
  "modified": "2020-10-02T01:29:11+00:00",
  "published": "2018-06-21T12:00:00+00:00",
  "aliases": [
    "CVE-2018-20989"
  ],
  "related": [],
  "package": {
    "ecosystem": "crates.io",
    "name": "untrusted",
    "purl": "pkg:cargo/untrusted"
  },
  "summary": "An integer underflow could lead to panic",
  "details": "A mistake in error handling in untrusted before 0.6.2 could lead to an integer\nunderflow and panic if a user of the crate didn't properly check for errors\nreturned by untrusted.\n\nCombination of these two programming errors (one in untrusted and another by\nuser of this crate) could lead to a panic and maybe a denial of service of\naffected software.\n\nThe error in untrusted is fixed in release 0.6.2 released 2018-06-21. It's also\nadvisable that users of untrusted check for their sources for cases where errors\nreturned by untrusted are not handled correctly.",
  "affects": {
    "ranges": [
      {
        "type": "SEMVER",
        "fixed": "0.6.2"
      }
    ]
  },
  "references": [
    {
      "type": "PACKAGE",
      "url": "https://crates.io/crates/untrusted"
    },
    {
      "type": "ADVISORY",
      "url": "https://rustsec.org/advisories/RUSTSEC-2018-0001.html"
    },
    {
      "type": "WEB",
      "url": "https://github.com/briansmith/untrusted/pull/20"
    }
  ],
  "ecosystem_specific": {
    "affects": {
      "arch": [],
      "os": [],
      "functions": []
    }
  },
  "database_specific": {
    "categories": [],
    "cvss": null,
    "informational": null
  }
}